src/app/models/field/emailField.ts

```diff
@@ -39,6 +39,41 @@ const createEmailFieldSchema = () => {
       type: Boolean,
       default: false,
     },
+    hasAllowedEmailDomains: {
+      type: Boolean,
+      default: false,
+    },
+    allowedEmailDomains: {
+      type: [String],
+      required: false,
+      // If there allowedEmailDomains is empty, then all email domains should be allowed.
+      default: [],
+      validate: {
+        validator: (emailDomains) => {
+          return (
+            !emailDomains.length ||
```

Comment:
If the validation is done with the individual match clause like above, this check can be simplified:

```ts
validator: (emailDomains: string[]) => {
    if (emailDomains.length === 0) return true
    return new Set(emailDomains).size === emailDomains.length
}
```